File: Myecoprint_admin/ncej_admin/lib/data/module/point.dart

class Point {
  int pointId;
  int storeId;
  double? totalPoints;
  int reasonBarcodeId;
  int? userId;


  Point({
    required this.pointId,
    required this.storeId,
    this.totalPoints,
    required this.reasonBarcodeId,
    this.userId,

  });

  factory Point.fromJson(Map<String, dynamic> json) {
    return Point(
      pointId: json['point_id']as int? ??0,
      storeId: json['store_id']as int? ??0,
      totalPoints: json['total_points'] != null
          ? json['total_points'].toDouble()
          : null,
      reasonBarcodeId: json['reasonBarcode_id']as int? ??0,
      userId: json['user_id']as int? ??0,

    );
  }

  Map<String, dynamic> toJson() {
    return {
      'point_id': pointId,
      'store_id': storeId,
      'total_points': totalPoints,
      'reasonBarcode_id': reasonBarcodeId,
      'user_id': userId,
    
    };
  }
}

File: backend_Myecoprint/backend_myecoprint/models/point.js

const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');
const reasonBarcode = require('./reason_barcode');
const User=require('./user')
const Store= require('./store');
const Point = sequelize.define('Point', {
    point_id: {
    type: DataTypes.INTEGER,
    allowNull: false,
    primaryKey: true,
    autoIncrement: true,
  },
  
store_id: {
    type: DataTypes.INTEGER,
    
  },
 last_daily_point:{
 type:DataTypes.DATE

 },
  total_points: {
    type: DataTypes.NUMBER,
  },
  reasonBarcode_id: {
    type: DataTypes.INTEGER,
  
  },
  user_id: {
    type: DataTypes.INTEGER,
  },
  createdAt: {
    type: DataTypes.DATE,
    allowNull: true,
  },
  updatedAt: {
    type: DataTypes.DATE,
    allowNull: true,
  },
  deletedAt: {
    type: DataTypes.DATE,
    allowNull: true,
  },
}, {
    timestamps: false, 
    paranoid: true, 
    tableName: 'points', 
  });
  Point.belongsTo(User, { foreignKey: 'user_id', as: 'users',onDelete: 'CASCADE' });
  Point.belongsTo(Store, { foreignKey: 'store_id', as: 'stores',onDelete: 'CASCADE' });
  Point .belongsTo(reasonBarcode, { foreignKey: 'reasonBarcode_id', as: 'reason_barcodes',onDelete: 'CASCADE' });

module.exports = Point;


























